keras_core/layers/preprocessing/random_rotation.py

```diff
@@ -76,52 +68,186 @@ class RandomRotation(Layer):
+        input_shape = len(inputs.shape)
+        if input_shape == 4:
+            if self.data_format == "channels_last":
+                batch_size = inputs.shape[0]
+                image_height = inputs.shape[1]
+                image_width = inputs.shape[2]
+            else:
+                batch_size = inputs.shape[1]
+                image_height = inputs.shape[2]
+                image_width = inputs.shape[3]
+        else:
+            batch_size = 1
+            if self.data_format == "channels_last":
+                image_height = inputs.shape[0]
+                image_width = inputs.shape[1]
+            else:
+                image_height = inputs.shape[1]
+                image_width = inputs.shape[2]
+
+        lower = self._factor[0] * 2.0 * self.backend.convert_to_tensor(np.pi)
+        upper = self._factor[1] * 2.0 * self.backend.convert_to_tensor(np.pi)
+
+        seed_generator = self._get_seed_generator(self.backend._backend)
+        angle = self.backend.random.uniform(
+            shape=(batch_size,),
+            minval=lower,
+            maxval=upper,
+            seed=seed_generator,
         )
-        self.supports_jit = False
-        self._convert_input_args = False
-        self._allow_non_tensor_positional_args = True
 
-    def call(self, inputs, training=True):
-        if not isinstance(inputs, (tf.Tensor, np.ndarray, list, tuple)):
-            inputs = tf.convert_to_tensor(backend.convert_to_numpy(inputs))
-        outputs = self.layer.call(inputs, training=training)
-        if (
-            backend.backend() != "tensorflow"
-            and not backend_utils.in_tf_graph()
-        ):
-            outputs = backend.convert_to_tensor(outputs)
+        cos_theta = self.backend.numpy.cos(angle)
+        sin_theta = self.backend.numpy.sin(angle)
+
+        x_offset = (
+            (image_width - 1)
+            - (cos_theta * (image_width - 1) - sin_theta * (image_height - 1))
+        ) / 2.0
+
+        y_offset = (
+            (image_height - 1)
+            - (sin_theta * (image_width - 1) + cos_theta * (image_height - 1))
+        ) / 2.0
+
+        outputs = self.backend.numpy.concatenate(
+            [
+                self.backend.numpy.cos(angle)[:, None],
+                -self.backend.numpy.sin(angle)[:, None],
+                x_offset[:, None],
+                self.backend.numpy.sin(angle)[:, None],
+                self.backend.numpy.cos(angle)[:, None],
+                y_offset[:, None],
+                self.backend.numpy.zeros((batch_size, 2), dtype=np.float32),
+            ],
+            axis=1,
+        )
+        if input_shape == 3:
+            outputs = self.backend.numpy.squeeze(outputs, axis=0)
         return outputs
 
+    def call(self, inputs, training=True):
+        inputs = self.backend.cast(inputs, self.compute_dtype)
+        if training:
+            rotation_matrix = self._get_rotation_matrix(inputs)
+            transformed_image = affine_transform(
```

Comment:
The failed tests should be fixed by replacing `affine_transform` from keras_core.ops.image with `self.backend.image.affine_transform`

Reply:
Hey @james77777778, thanks for the help!!

Reply:
I'm getting a really small difference in the element which is supposed to be 0, in the torch implementation, about 2.62260437e-06. Is it wise to increase rtol?
Also this is after importing the torch affine transform function updation.
What do you suggest I do?
cc: @fchollet

Reply:
> Is it wise to increase rtol?

Yes, any numerical difference that is close to the float32 precision is fine to write off. You can use a 1e-5 rtol.